azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -79,9 +79,27 @@ def is_policy_enforcement_enabled():
 
     def __get_policy(self):
         """
-         Check if custom policy exists at CUSTOM_POLICY_PATH, load JSON object and return as a dict.
-        Return {} if no policy exists.
-        The expected policy format is:
+        Check if custom policy exists at CUSTOM_POLICY_PATH, load JSON object and return as a dict.
+        Return default policy if no policy exists.
+        """
+        # TODO: Add schema validation for custom policy file and raise relevant error message to user.
+        if os.path.exists(CUSTOM_POLICY_PATH):
+            self._log_policy_event("Custom policy found at {0}. Using custom policy.".format(CUSTOM_POLICY_PATH))
+            with open(CUSTOM_POLICY_PATH, 'r') as f:        # Open file in read-only mode.
+                custom_policy = json.load(f)
```

Comment:
Same comment as before here - 
What happens if policy file content is not valid json? Does this raise exception? How do we want to handle that right now

Reply:
If the policy file is not a valid json, this raises JSONDecodeError. I added a unit test for this specific case.

Reply:
Could you point me to the unit test?

Where will this exception be handled?

Reply:
Discussed offline - I will add "PolicyInvalidError" class specifically for policy format/syntax errors with a clear error message for the user

Reply:
Unit test is called "test_should_raise_value_error_if_policy_file_is_invalid_json"